hack/check-pr.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: check-pr [ref]
+#
+# Ideally if check-pr passes then your golang PR will pass presubmit tests.
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+dirs=()
+tests=()
```

Comment:
why not let the bazel cache handle this? If we test the dirs that changed we can miss EG changing a package depended on by tests in many other dirs.

Reply:
Two primary reasons:
a) `bazel test //...` is no longer working on my laptop, I keep getting `go/src/k8s.io/test-infra/.git/logs/refs/heads/BUILD:4:101: invalid character: '@'` errors (`hack/build-then-unit.sh` also fails)
b) `bazel test //...` takes forever on my laptop

Most of the time testing the changed packages should work well enough. Failures can get caught on presubmit. And if it happens enough times I'm sure I'll update the

This is mostly to help me do the gofmt, govet, golint, verify-bazel check in one go. Cause I'm constantly forgetting one of the checks and have to send 5 commits to the PR before I get everything working correctly 😅

Reply:
Fair, I think we should fix a) though, taking a look. 

b) should be a one-time cost and is also because some of our tests take longer than they should. ... I think we should also consider fixing that.. Seems more orthogonal though.

Reply:
"one-time cost" aka what feels like once per rebase and/or branch